pkg/table/tables/partition.go

```diff
@@ -1801,17 +1811,16 @@ func partitionedTableUpdateRecord(gctx context.Context, ctx table.MutateContext,
 			tbl := t.GetPartition(newTo)
 			return tbl.UpdateRecord(gctx, ctx, h, currData, newData, touched)
```

Comment:
What happens if an error is returned here? Will the deferred function still do the right thing?

Reply:
Updated